pagination/README.md

```diff
@@ -0,0 +1,100 @@
+## Package `pagination` v0
+
+This is a pagination library for Go.
+
+This library provides a data structure to transport paginated data. It is useful when you need to return a page of items
+along with the total number of items and other Metadata.
+
+**What this library does:**
+
+* it provides a data structure to transport paginated data.
+* it provides functions to create "paginated data" and "page request" structures.
+* it accepts data as a slice of any type. The data supplied must be 1 page of items.
+
+**What this library does NOT do:**
+
+* it is NOT responsible for fetching the data from the database.
+* it is NOT responsible for splitting a result set into pages.
+
+The `PageRequest` struct is used to define the page number and the number of items per page. Used when requesting a page
+of items.
+
+The `Paginated` struct is used to return a page of items along with the total number of items.
+
+// Example data returned
+
+```json 
+{
+  "data": [],
+  "meta": {
+    "current_page": 1,
+    "next_page": 1,
+    "per_page": 20,
+    "prev_page": 1,
+    "total_count": 0,
+    "total_pages": 1
+  }
+}
+```
+
+### Usage
+
+#### Example of the first page of data of a small dataset:
+
+In this example, we create a new `Paginated` struct with a slice of `Item` and a `PageRequest` struct.
+
+* The total number of items in the database is 4.
+* The `PageRequest` struct specifies that we want the first page with 10 items per page.
+* The `items` slice contains 4 items (the same as the number of total items, but less than a full page).
```

Comment:
praise: thanks for this introduction to explain the example. It helps a lot!
suggestion: don't you think it would be clearer if we display 2 items per page in the example so that it's an example with multiple pages?